### ISSUE TITLE
Add 2016 election precinct shapes for Kansas

### ISSUE THREAD

**Issue by nvkelso** [nvkelso/election-geodata#9]
This state is in priority tier 1 (out of all US states) according to https://lawreview.uchicago.edu/sites/lawreview.uchicago.edu/files/uploads/82_2/04%20Stephanopoulos_McGhee_ART.pdf (via http://mike.teczno.com/notes/redistricting/measuring-efficiency-gap.html).

**105 counties.**
_Worse than California, only better than Texas, Georgia, Virginia, Kentucky, and Missouri. [src](https://en.wikipedia.org/wiki/County_statistics_of_the_United_States)_

- [ ] 20001 - KS statewide PR https://github.com/nvkelso/election-geodata/pull/60
---
- [ ] 20001 - Allen County, KS **(27)**
- [ ] 20003 - Anderson County, KS **(25)**
- [ ] 20005 - Atchison County, KS **(25)**
- [ ] 20007 - Barber County, KS **(24)**
- [ ] 20009 - Barton County, KS **(50)**
- [ ] 20011 - Bourbon County, KS **(30)**
- [ ] 20013 - Brown County, KS **(28)**
- [ ] 20015 - Butler County, KS **(81)**
- [ ] 20017 - Chase County, KS **(11)**
- [ ] 20019 - Chautauqua County, KS **(14)**
- [ ] 20021 - Cherokee County, KS **(54)**
- [ ] 20023 - Cheyenne County, KS **(8)**
- [ ] 20025 - Clark County, KS **(7)**
- [ ] 20027 - Clay County, KS **(25)**
- [ ] 20029 - Cloud County, KS **(29)**
- [ ] 20031 - Coffey County, KS **(23)**
- [ ] 20033 - Comanche County, KS **(4)**
- [ ] 20035 - Cowley County, KS **(69)**
- [ ] 20037 - Crawford County, KS **(62)**
- [ ] 20039 - Decatur County, KS **(28)**
- [ ] 20041 - Dickinson County, KS **(36)**
- [ ] 20043 - Doniphan County, KS **(13)**
- [x] 20045 - Douglas County, KS **(95)** PR #55 
- [ ] 20047 - Edwards County, KS **(13)**
- [ ] 20049 - Elk County, KS **(10)**
- [ ] 20051 - Ellis County, KS **(39)**
- [ ] 20053 - Ellsworth County, KS **(25)**
- [ ] 20055 - Finney County, KS **(32)**
- [ ] 20057 - Ford County, KS **(32)**
- [ ] 20059 - Franklin County, KS **(28)**
- [ ] 20061 - Geary County, KS **(61)**
- [ ] 20063 - Gove County, KS **(9)**
- [ ] 20065 - Graham County, KS **(17)**
- [ ] 20067 - Grant County, KS **(5)**
- [ ] 20069 - Gray County, KS **(8)**
- [ ] 20071 - Greeley County, KS **(3)**
- [ ] 20073 - Greenwood County, KS **(21)**
- [ ] 20075 - Hamilton County, KS **(10)**
- [ ] 20077 - Harper County, KS **(22)**
- [ ] 20079 - Harvey County, KS **(44)**
- [ ] 20081 - Haskell County, KS **(8)**
- [ ] 20083 - Hodgeman County, KS **(11)**
- [ ] 20085 - Jackson County, KS **(20)**
- [ ] 20087 - Jefferson County, KS **(13)**
- [ ] 20089 - Jewell County, KS **(25)**
- [ ] 20091 - Johnson County, KS **(444)**
- [ ] 20093 - Kearny County, KS **(9)**
- [ ] 20095 - Kingman County, KS **(30)**
- [ ] 20097 - Kiowa County, KS **(5)**
- [ ] 20099 - Labette County, KS **(42)**
- [ ] 20101 - Lane County, KS **(10)**
- [ ] 20103 - Leavenworth County, KS **(52)**
- [ ] 20105 - Lincoln County, KS **(24)**
- [ ] 20107 - Linn County, KS **(13)**
- [ ] 20109 - Logan County, KS **(12)**
- [ ] 20111 - Lyon County, KS **(53)**
- [ ] 20113 - McPherson County, KS **(34)**
- [ ] 20115 - Marion County, KS **(35)**
- [ ] 20117 - Marshall County, KS **(34)**
- [ ] 20119 - Meade County, KS **(9)**
- [ ] 20121 - Miami County, KS **(59)**
- [ ] 20123 - Mitchell County, KS **(33)**
- [ ] 20125 - Montgomery County, KS **(60)**
- [ ] 20127 - Morris County, KS **(19)**
- [ ] 20129 - Morton County, KS **(10)**
- [ ] 20131 - Nemaha County, KS **(38)**
- [ ] 20133 - Neosho County, KS **(31)**
- [ ] 20135 - Ness County, KS **(11)**
- [ ] 20137 - Norton County, KS **(12)**
- [ ] 20139 - Osage County, KS **(28)**
- [ ] 20141 - Osborne County, KS **(27)**
- [ ] 20143 - Ottawa County, KS **(24)**
- [ ] 20145 - Pawnee County, KS **(26)**
- [ ] 20147 - Phillips County, KS **(29)**
- [ ] 20149 - Pottawatomie County, KS **(30)**
- [ ] 20151 - Pratt County, KS **(17)**
- [ ] 20153 - Rawlins County, KS **(18)**
- [ ] 20155 - Reno County, KS **(80)**
- [ ] 20157 - Republic County, KS **(26)**
- [ ] 20159 - Rice County, KS **(28)**
- [ ] 20161 - Riley County, KS **(71)**
- [ ] 20163 - Rooks County, KS **(14)**
- [ ] 20165 - Rush County, KS **(13)**
- [ ] 20167 - Russell County, KS **(19)**
- [ ] 20169 - Saline County, KS **(64)**
- [ ] 20171 - Scott County, KS **(11)**
- [ ] 20173 - Sedgwick County, KS **(447)**
- [ ] 20175 - Seward County, KS **(23)**
- [ ] 20177 - Shawnee County, KS **(227)**
- [ ] 20179 - Sheridan County, KS **(15)**
- [ ] 20181 - Sherman County, KS **(19)**
- [ ] 20183 - Smith County, KS **(27)**
- [ ] 20185 - Stafford County, KS **(24)**
- [ ] 20187 - Stanton County, KS **(3)**
- [ ] 20189 - Stevens County, KS **(11)**
- [ ] 20191 - Sumner County, KS **(47)**
- [ ] 20193 - Thomas County, KS **(23)**
- [ ] 20195 - Trego County, KS **(11)**
- [ ] 20197 - Wabaunsee County, KS **(14)**
- [ ] 20199 - Wallace County, KS **(5)**
- [ ] 20201 - Washington County, KS **(29)**
- [ ] 20203 - Wichita County, KS **(3)**
- [ ] 20205 - Wilson County, KS **(29)**
- [ ] 20207 - Woodson County, KS **(11)**
- [ ] 20209 - Wyandotte County, KS **(111)**

**Comment by nvkelso** [comment]
When researching & requesting data also ask for **matching election results** (_presidential, congressional, state house, and state senate_) in addition to the precinct shape data for 2016, **and for earlier election years** if available.

**Comment by nvkelso** [comment]
Looks like incomplete county based precinct coverage at Open Elections:

- https://github.com/openelections/openelections-data-ks/tree/master/2016
- https://github.com/openelections/openelections-data-ks/tree/master/2014

**Comment by karpet** [comment]
Great project. I am investigating KS precinct GeoJSON data, which appears to be necessary county-by-county. Starting now with Douglas County.

**Comment by nvkelso** [comment]
@karpet Thank you! I've updated the issue description to include checkboxes for the 105 counties of Kansas and their FIPS codes ([src](https://www.nrcs.usda.gov/wps/portal/nrcs/detail/ks/home/?cid=nrcs143_013697)). 

It looks like US Census has a _list_ of voting precincts (last decennial in 2010) in all of Kansas to give us a rough idea of total coverage count. There are 3907 in that file (linked below).

The Kansas precincts mostly have sane names (versus California where most are just numbered), I wonder if they are reverse geocode-able? Many look to correspond to `localadmin` areas, of which Who's On First (via US Census) has [1530 count](https://whosonfirst.mapzen.com/spelunker/id/85688555/descendants/?exclude=nullisland&placetype=localadmin#7/38.502/-99.980) and polygons (which are [downloadable](https://whosonfirst.mapzen.com/spelunker/download/85688555/?exclude=nullisland#6/38.514/-98.320)). 

When I do some quick text analysis it look like a pretty good match, with the other ~1500 precincts being in urbanized city areas where we could at least get a rough "city" lat/lng point, and a more specific polygon (or centroid) for the townships. This iterative approach might be useful to prioritize which counties have more-or-less complete township === precinct data, versus mostly urban counties where the more detailed precincts would be useful. Population count is also good priority proxy.

@migurski do you need polygons or would lat/lng points work in  a pinch? Over time more would be polygons, but looks like right now we could do 50% of one, 50% of the other.

- http://www2.census.gov/geo/docs/reference/codes/files/st20_ks_vtd.txt
- from: https://www.census.gov/geo/reference/codes/vtd.html

```
STATE|STATEFP|COUNTYFP|COUNTYNAME|VTDST|VTDNAME
KS|20|145|Pawnee County|000050|Grant Township Voting District
KS|20|145|Pawnee County|000060|Keysville Township Voting District
KS|20|145|Pawnee County|000070|Larned Township Voting District
KS|20|145|Pawnee County|000080|Larned Ward 1
KS|20|145|Pawnee County|000090|Larned Ward 2
KS|20|145|Pawnee County|000100|Larned Ward 3
KS|20|145|Pawnee County|000110|Larned Ward 4
KS|20|145|Pawnee County|000120|Lincoln Township Voting District
KS|20|145|Pawnee County|000130|Logan Township Voting District
KS|20|145|Pawnee County|000140|Morton Township Voting District
```

**Comment by migurski** [comment]
I think points could work — we could use them to build a Voronoi tesselation as a cheap starter set of polygons, and then refine if more data becomes available? Really curious if @anthea is still present on Github and has something to say about this.

**Comment by nvkelso** [comment]
Using Pivot table in OpenOffice I updated the issue description to include precinct counts per county (but kept that list alphabetical).

Here are the counties with 50 or more precincts and where I'd prioritize work:

FIPS   | County             | Precincts
------ | ------------------ | ---------
20173 | Sedgwick County, KS | 447
20091 | Johnson County, KS | 444
20177 | Shawnee County, KS | 227
20209 | Wyandotte County, KS | 111
20045 | Douglas County, KS | 95
20015 | Butler County, KS | 81
20155 | Reno County, KS | 80
20161 | Riley County, KS | 71
20035 | Cowley County, KS | 69
20169 | Saline County, KS | 64
20037 | Crawford County, KS | 62
20061 | Geary County, KS | 61
20125 | Montgomery County, KS | 60
20121 | Miami County, KS | 59
20021 | Cherokee County, KS | 54
20111 | Lyon County, KS | 53
20103 | Leavenworth County, KS | 52
20009 | Barton County, KS | 50

**Comment by karpet** [comment]
@nvkelso I got the ERSI geodata file from the county, converted it to PostGIS, and then exported it as GeoJSON. Not sure if it's the format you're looking for though. Happy to transform it into something compatible if you can point me at a schema or example.

https://github.com/karpet/election-geodata/tree/master/data/20-kansas/20045-douglas

**Comment by nvkelso** [comment]
For those like me catching on slowly, Douglas Co = Lawrence. Kansas :)

I loaded your GeoJSON into QGIS and it shows up in the right spot, with
good attributes. Works for me!

For some reason Github won't preview the GeoJSON file, but the validation
in QGIS shows it's good enough.

Want to file a PR for this one from your fork?

**Comment by karpet** [comment]
Yup #55

**Comment by nvkelso** [comment]
Progress!

![image](https://cloud.githubusercontent.com/assets/853051/23681358/9170fc72-0343-11e7-940b-6f45ff8a00c7.png)

**Comment by karpet** [comment]
I was able to locate statewide districts for 2010. Anything newer will require going county-by-county, but I am discovering the GIS community in KS is happy to help when asked. Ever forward.

See #60

**Comment by migurski** [comment]
So glad to hear it, @karpet!

**Comment by nvkelso** [comment]
Looks like 2012 precincts may also be available:

- https://github.com/aaron-strauss/precinct-shapefiles/tree/master/ks

**Comment by karpet** [comment]
Confirmed.

http://kansasgis.org/catalog/index.cfm?data_id=2081&show_cat=1?data_id=2081&show_cat=1

I will add to #60

**Comment by nvkelso** [comment]
Closed #9 via https://github.com/nvkelso/election-geodata/pull/60.
